packages/common/src/stream.ts

```diff
@@ -118,6 +118,11 @@ export interface LogEntry {
   cid: CID
   type: CommitType
   timestamp?: number
+  // Holding on to these fields when fetching and processing new commits significantly reduces the number of IPFS
+  // lookups required. Use of these fields is currently limited to `conflict-resolution.ts` but further optimization of
+  // the code is possible by relaying them to code that looks the fields up again.
+  commit?: any
+  envelope?: DagJWS
```

Comment:
Why both `commit` and `envelope`?

Reply:
Yes, the `commit` would always be a plain raw or anchor commit. In the case of a `SignedCommit`, it will be the linked commit. The `envelope` would be the DagJWS, in case of a `SignedCommit`.

It made sense to separate those two concepts versus considering them all under the same term. Spencer did suggest calling it `signature` instead of `envelope` but I thought the former has a more specific meaning in the code as well.